### PR TITLE
Implement expand_dims

### PR DESCRIPTION
**Pull request by jeanlaroche**
np.expand_dims was missing...

### CODE REVIEW

File: pythran/pythonic/numpy/expand_dims.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  template <typename T>
+  types::ndarray<typename T::dtype, types::array<long, T::value + 1>>
+  expand_dims(T const &input, int axis)
+  {
+    const int N = std::tuple_size<decltype(input._shape)>::value;
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/expand_dims.hpp]: could be `const long N = T::value`

File: pythran/pythonic/numpy/expand_dims.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    auto in_shape = sutils::array(input._shape);
+    long ii, jj;
+    for (ii = jj = 0; ii < N + 1; ii++) {
+      if (ii == axis) {
+        dim_array[ii] = 1;
+      } else {
+        dim_array[ii] = in_shape[jj++];
+      }
+    }
+    sutils::push_front_t<decltype(input._shape), long> new_shape;
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/expand_dims.hpp]: this is incorrect, as your return type specifies ` types::array<long, T::value + 1>` as return type. why not reshaping directly using `dim_array` ?

File: pythran/pythonic/numpy/expand_dims.hpp (hunk 1 above)
Comment (jeanlaroche) [reply pythran/pythonic/numpy/expand_dims.hpp]: ah ah :) Thanks! There had to be a simpler way. You can see I'm still not familiar with the classes used in pythran!

File: pythran/pythonic/numpy/expand_dims.hpp (hunk 2 above)
Comment (jeanlaroche) [reply pythran/pythonic/numpy/expand_dims.hpp]: Yes, much simpler, I didn't realize I could pass an array, instead of a pshape.

Comment (jeanlaroche) [reply pythran/pythonic/numpy/expand_dims.hpp]: Serge, if I try reshaping with
```
return ndarray::reshape(input, dim_array);
```
compilation fails. I'm having a hard time figuring out whether reshape can accept an array as input....

**Pull request merged**